cpp/src/groupby/groupby.cu

```diff
@@ -219,20 +219,18 @@ std::pair<std::unique_ptr<table>, std::vector<aggregation_result>> groupby::scan
 groupby::groups groupby::get_groups(table_view values, rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  auto grouped_keys = helper().sorted_keys(rmm::cuda_stream_default, mr);
+  auto const stream = rmm::cuda_stream_default;
```

Comment:
Why don't we make `stream` as a function parameter (with default value)?

Reply:
That would apply to several functions in this file, which declare that they are using the default stream in the function but do not accept an argument for it. I’m not sure if I was ready to make that API change in the scope of this PR, but this PR is a step in that direction.

Reply:
My primary goal was to use a vector factory instead of a thrust call and keep the PR small / non-breaking.

Reply:
I'm not sure I understand why adding a parameter that has a default would impact other places.

Reply:
This is a public API. We do not currently expose streams on public APIs.

Reply:
Thanks @jrhemstad — I meant to check if it was public and forgot since I was commenting from my phone. Based on that, I recommend we resolve this and accept the change as-is.